plenum/test/client/test_core_authn.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+
+from plenum.common.constants import GET_TXN, NODE, NYM
+from plenum.common.exceptions import CouldNotAuthenticate, \
+    InsufficientSignatures, InsufficientCorrectSignatures, MissingSignature
+from plenum.common.signer_simple import SimpleSigner
+from plenum.common.types import f
+from plenum.server.client_authn import CoreAuthNr
+
+
+idr = '5G72199XZB7wREviUbQma7'
+msg_str = "42 (forty-two) is the natural number that succeeds 41 and precedes 43."
+
+
+class DummyAuthenticator(CoreAuthNr):
+    def getVerkey(self, _):
+        return None
+
+
+@pytest.fixture(scope="module")
+def cli():
+    return SimpleSigner(idr)
+
+
+@pytest.fixture(scope="module")
+def sa(cli):
+    sa = CoreAuthNr()
+    sa.addIdr(cli.identifier, cli.verkey)
+    return sa
+
+
+@pytest.fixture(scope="module")
+def msg():
+    return {'myMsg': msg_str, f.IDENTIFIER.nm: idr}
+
+
+@pytest.fixture(scope="module")
+def sig(cli, msg):
+    return cli.sign(msg)
+
+
+def test_authenticate_raises_correct_exception():
+    msg = dict(myMsg=msg_str)
+    simple_signer = SimpleSigner()
+    identifier = simple_signer.identifier
+    signature = simple_signer.sign(msg)
```

Comment:
Why do we sign message and expect `MissingSignature` ?

Reply:
Because signature is not passed as an argument

Reply:
Because signature is not passed as an argument

Reply:
Then we can not call `sign` at all.